DSI/globus_gridftp_server_iRODS.cpp

```diff
@@ -99,12 +100,54 @@ extern "C" {
 /* if present, set the number or read/write threads for file transfers */
 #define IRODS_PARALLEL_FILE_SIZE_THRESHOLD_BYTES "irodsParallelFileSizeThresholdBytes"
 
-
 const static unsigned int DEFAULT_NUMBER_OF_IRODS_READ_WRITE_THREADS = 3;
 const static unsigned int MAXIMUM_NUMBER_OF_IRODS_READ_WRITE_THREADS = 10;
 
 const static std::string CHECKSUM_AVU_NAMESPACE{"GLOBUS"};
 
+// map to translate globus_gfs_command_type_t values into strings for error messages
+static std::map<int, std::string> command_map = {
+    { 1, "GLOBUS_GFS_CMD_MKD" },
+    { 2, "GLOBUS_GFS_CMD_RMD" },
+    { 3, "GLOBUS_GFS_CMD_DELE" },
+    { 4, "GLOBUS_GFS_CMD_SITE_AUTHZ_ASSERT" },
+    { 5, "GLOBUS_GFS_CMD_SITE_RDEL" },
+    { 6, "GLOBUS_GFS_CMD_RNTO" },
+    { 7, "GLOBUS_GFS_CMD_RNFR" },
+    { 8, "GLOBUS_GFS_CMD_CKSM" },
+    { 9, "GLOBUS_GFS_CMD_SITE_CHMOD" },
+    { 10, "GLOBUS_GFS_CMD_SITE_DSI" },
+    { 11, "GLOBUS_GFS_CMD_SITE_SETNETSTACK" },
+    { 12, "GLOBUS_GFS_CMD_SITE_SETDISKSTACK" },
+    { 13, "GLOBUS_GFS_CMD_SITE_CLIENTINFO" },
+    { 14, "GLOBUS_GFS_CMD_DCSC" },
+    { 15, "GLOBUS_GFS_CMD_SITE_CHGRP" },
+    { 16, "GLOBUS_GFS_CMD_SITE_UTIME" },
+    { 17, "GLOBUS_GFS_CMD_SITE_SYMLINKFROM" },
+    { 18, "GLOBUS_GFS_CMD_SITE_SYMLINK" },
+    { 19, "GLOBUS_GFS_CMD_HTTP_PUT" },
+    { 21, "GLOBUS_GFS_CMD_HTTP_GET" },
+    { 22, "GLOBUS_GFS_CMD_HTTP_CONFIG" },
+    { 23, "GLOBUS_GFS_CMD_TRNC" },
+    { 24, "GLOBUS_GFS_CMD_SITE_TASKID" },
+    { 3072, "GLOBUS_GFS_CMD_SITE_RESTRICT" },
+    { 3073, "GLOBUS_GFS_CMD_SITE_CHROOT" },
+    { 3074, "GLOBUS_GFS_CMD_SITE_SHARING" },
+    { 3075, "GLOBUS_GFS_CMD_UPAS" },
+    { 3076, "GLOBUS_GFS_CMD_UPRT" },
+    { 3077, "GLOBUS_GFS_CMD_STORATTR" },
+    { 3078, "GLOBUS_GFS_CMD_WHOAMI" },
+    { 4096, "GLOBUS_GFS_MIN_CUSTOM_CMD" }
+};
+
+const std::string get_command_string(const int i) {
```

Comment:
`const std::string` -> `const std::string&`?

Reply:
I can't because it returns a std::to_string(i) if the lookup fails.

Reply:
Oh right. Good eye.